### PR TITLE
Add facebook ref code, get to 100% coverage

### PR DESCRIPTION
**Pull request by nicpottier**
Need the facebook ref stuff for courier dealing with mapping these back to real facebook ids. Decided to just bump coverage to 100% while I was at it.

### CODE REVIEW

File: urns/urns.go [hunk 1]
```diff
@@ -237,9 +240,24 @@ func (u URN) Localize(country string) URN {
...
+	if u.IsFacebookRef() {
+		return strings.TrimPrefix(u.Path(), FacebookRefPrefix)
+	}
+	return ""
+}
+
 // Resolve is called when a URN is part of an excellent expression
 func (u URN) Resolve(key string) interface{} {
 	switch key {
+	case "display":
```
Comment (nicpottier) [inline urns/urns.go]: For completeness.

File: urns/urns.go [hunk 2]
```diff
@@ -254,7 +272,7 @@ func (u URN) Resolve(key string) interface{} {
 func (u URN) Default() interface{} { return u }
 
 // String returns the string representation of this URN
-func (u URN) String() string { return string(u.Path()) }
+func (u URN) String() string { return string(u) }
```
Comment (nicpottier) [inline urns/urns.go]: Sure seems like the string value for a URN should be the full representation for the URN. This kind of maps to the values for locations and times being fully representative and something like `format_urn` being responsible for creating user friendly versions.

Comment (rowanseymour) [reply urns/urns.go]: Yeah this a product of not yet having a format_urn and wanting contact.urns.tel to resolve to telephone number - def agree this should be the whole thing.

File: urns/urns_test.go [hunk 3]
```diff
@@ -144,6 +218,12 @@ func TestValidate(t *testing.T) {
 		{"telegram:abcdef", false},
 		{"facebook:12345678901234567", true},
 		{"facebook:abcdef", false},
+
+		// facebook refs can be anything
+		{"facebook:ref:facebookRef", true},
```
Comment (rowanseymour) [inline urns/urns_test.go]: too late now but wish we were using something else between `ref` and the rest of the path

Comment (nicpottier) [reply urns/urns_test.go]: I think its actually URNy to use `:`s in this way but agree its kinda weird

**Pull request merged**